hyperdash/experiment.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from .client import HDClient
+from .monitor import monitor
+from .io_buffer import IOBuffer
+from .server_manager import ServerManagerHTTP
+from .hyper_dash import HyperDash
+from .utils import get_logger
+
+import sys
+import uuid
+import threading
+from six.moves.queue import Queue
+# Python 2/3 compatibility
+__metaclass__ = type
+
+class ExperimentRunner:
+    def __init__(
+        self,
+        done=False,
+        exit_cleanly=True,
+    ):
+        self.done = done
```

Comment:
Look at this file: https://github.com/hyperdashio/hyperdash-sdk-py/blob/master/hyperdash/code_runner.py to see how to add a super simple lock